presto-main/src/main/java/com/facebook/presto/sql/planner/LogicalPlanner.java

```diff
@@ -163,6 +163,12 @@ private RelationPlan createExplainAnalyzePlan(Analysis analysis, Explain stateme
     {
         Scope scope = analysis.getScope(statement);
         PlanNode root = planStatement(analysis, statement.getStatement());
+        if (root instanceof OutputNode) {
+            // this extra output node was added by a call to plan underlying query and is not needed
```

Comment:
Where did the bad `OutputNode` come from? It looks to me like it's only constructed in `createOutputPlan` and there it uses the `RelationType` to create its outputs